src/Clapi/Relay.hs

```diff
@@ -209,3 +225,24 @@ relay vs = waitThen fwd (const $ error "message from the void")
         sendRev $ ServerData awu ms'
         relay vs'
     fwd _ = relay vs
+
+data NsChange =
+    NsAssign Name Path
+  | NsRemove Name
+
+nsChange :: VsDelta -> Maybe NsChange
+nsChange (p, Left tp) = if isNamespace p then Just $ NsAssign (head p) tp else Nothing
+nsChange (p, Right Tree.Delete) = if isNamespace p then Just $ NsRemove (head p) else Nothing
+nsChange _ = Nothing
+
+updateRootType :: [ClapiValue] -> NsChange -> [ClapiValue]
+updateRootType [lib, doc, ClList names, ClList types, ClList libs] m = lib:doc:tinfo
+  where
+    tinfo = map ClList $ case m of
+        (NsAssign n tp) -> [c n:names, cp tp:types, cannot:libs]
+        (NsRemove n) -> l3 . unzip3 $ filter (nameIsnt n) $ zip3 names types libs
+    l3 (ns, ts, ls) = [ns, ts, ls]
+    nameIsnt n (n', _, _) = n' /= c n
+    c = ClString . T.pack
+    cp p = c $ concat $ intersperse "/" p
+    cannot = ClEnum $ fromIntegral $ fromEnum $ Cannot
```

Comment:
See `Clapiable` instance for enumerated types. You should just be able to use `toClapiValue Cannot` here!

Reply:
It comes out as `toClapiValue $ Enumerated Cannot` presumably to avoid issues with Bool.

Reply:
Yeah, I'd forgotten we have a wrapper.